packages/commonwealth/client/scripts/state/api/communities/getCommuityById.ts

```diff
@@ -0,0 +1,92 @@
+import { ExtendedCommunity } from '@hicommonwealth/schemas';
+import axios from 'axios';
+import { trpc } from 'utils/trpcClient';
+import { z } from 'zod';
+import { queryClient, SERVER_URL } from '../config';
+
+const COMMUNITIY_STALE_TIME = 60 * 3_000; // 3 mins
+
+type UseGetCommunityByIdProps = {
+  id: string;
+  includeNodeInfo?: boolean;
+  enabled?: boolean;
+};
+
+export const invalidateAllQueriesForCommunity = async (communityId: string) => {
+  // get all the query keys for this community
+  const queryKeys = [
+    trpc.community.getCommunity.getQueryKey({
+      id: communityId,
+      include_node_info: true,
+    }),
+    trpc.community.getCommunity.getQueryKey({
+      id: communityId,
+      include_node_info: false,
+    }),
+  ];
+
+  // invalidate all the query keys for this community
+  if (queryKeys.length > 0) {
+    await Promise.all(
+      queryKeys.map(async (key) => {
+        const params = { queryKey: key };
+        await queryClient.cancelQueries(params);
+        await queryClient.invalidateQueries(params);
+      }),
+    );
+  }
+};
+
+export const EXCEPTION_CASE_VANILLA_getCommunityById = async (
+  communityId: string,
+  includeNodeInfo = false,
+): Promise<z.infer<typeof ExtendedCommunity> | undefined> => {
+  // make trpc query key for this request
+  const queryKey = trpc.community.getCommunity.getQueryKey({
+    id: communityId,
+    include_node_info: includeNodeInfo,
+  });
+
+  // if community already exists in cache, return that
+  const cachedCommunity = queryClient.getQueryData<
+    z.infer<typeof ExtendedCommunity> | undefined
+  >(queryKey);
+  if (cachedCommunity) {
+    return cachedCommunity;
+  }
+
+  // HACK: with @trpc/react-query v10.x, we can't directly call an endpoint outside of 'react-context'
+  // with this way the api can be used in non-react files. This should be cleaned up when we migrate
+  // to @trpc/react-query v11.x
```

Comment:
can you create ticket for this migration

Reply:
@trpc/react-query v11.x is still with the @next flag (active development + can have breaking changes) -- not sure if migrating now would be a good idea.

Will create a ticket to clean this up when we eventually migrate to @trpc/react-query v11.x